src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -31,7 +32,27 @@ public class CompareSymbolsCorrectlyAnalyzer : DiagnosticAnalyzer
             description: s_localizableDescription,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+        public static readonly DiagnosticDescriptor GetHashCodeRule = new DiagnosticDescriptor(
+            DiagnosticIds.CompareSymbolsCorrectlyRuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: s_localizableDescriptionGetHashCode,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public static readonly DiagnosticDescriptor CollectionRule = new DiagnosticDescriptor(
+            DiagnosticIds.CompareSymbolsCorrectlyRuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: s_localizableDescription,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(EqualityRule);
```

Comment:
I haven't put the other descriptors here because otherwise I have to change all use of the markup syntax in the tests. @sharwell Maybe the check shall be relaxed to allow markup syntax when the different descriptors registered have the same ID, WDYT?

Reply:
I think we should make different ID for the checks. Even if messaging remains the same, it will help correlate the rule being triggered.